src/compose/images.ts

```diff
@@ -598,12 +598,15 @@ export async function inspectByName(imageName: string) {
 
 	// Run the queries in sequence, return the first one that matches or
 	// the error from the last query
-	return await [inspectByURI, inspectByReference, inspectByDigest].reduce(
-		(promise, query) => promise.catch(() => query(imageName)),
-		Promise.reject(
-			'Promise sequence in inspectByName is broken. This is a bug.',
-		),
-	);
+	let err;
+	for (const query of [inspectByURI, inspectByReference, inspectByDigest]) {
+		try {
+			return await query(imageName);
```

Comment:
This will allow for better async stack traces by using `await` and I personally find the overall reformulation clearer to understand